x-pack/plugin/esql/build.gradle

```diff
@@ -227,7 +224,7 @@ tasks.register("regenParser", JavaExec) {
 }
 
 tasks.register("regen") {
-  dependsOn "regenParser"
+  dependsOn "regenParser", "stringTemplates"
```

Comment:
I had to ensure `stringTemplates` runs before other code generation.

This has to be done as `stringTemplates` task cleans the entire generated directory before execution (see [StringTemplateTask](https://github.com/elastic/elasticsearch/blob/a59c182f9f7e9d1bf3d6eecbc0e44f24ff91d053/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/StringTemplateTask.java#L69-L70)) while other generation not.

This means that before this change `/generated` dir was not cleaned before generating new files, keeping files with corresponding templates deleted.

Reply:
Are you sure this is needed? The `regen` task uses a different mechanism and doesn't put the output in the same dir. It makes Antlr stuff everything into `x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser`.

I just removed the `stringTemplates` dependency locally and made a small change to our grammar - the diff looked very sane and contained; whereas with the dependency on `stringTemplates` in place, making changes to the grammar and running `regen` deletes a bunch of unrelated files, which one must not commit. I think that'll make making changes to the grammar a bit painful.

Reply:
I tried to group it with other `gen` stuff I noticed, but may be it is worth to update the `stringTemplates` not to clear files unconditionally and instead have dedicated clear generated files tasks. What gradle task do we use to generate other files in `main/gnerated`?

Antlr generated code goes into `main/java`, I do not think this is intended as well. I will check If I can move it to generated in a separate pr.